MetaMask/metamask-extension
# Adding `BaseIcon` UI component

## Explanation

Adding `BaseIcon` component as a base for all icon components.

Another PR exists that includes a script to create react components from svgs https://github.com/MetaMask/metamask-extension/pull/15555 which this is a dependency of

## More Information

* Fixes #15598
* Apart of https://github.com/MetaMask/metamask-extension/issues/15098

## Screenshots/Screencaps

### After


https://user-images.githubusercontent.com/8112138/185239451-94866949-e5cd-4a64-898e-b568c076041c.mov



## Manual Testing Steps

- Go to the latest storybook build on this PR
<img width="420" alt="Screen Shot 2022-08-17 at 3 23 45 PM" src="https://user-images.githubusercontent.com/8112138/185254122-f016aa21-e528-435a-9aa3-af8343f5fa8c.png">

- Search `BaseIcon` in the search bar
- Check controls, stories and documentation

## Pre-Merge Checklist

- ~[x] PR is linked to the appropriate GitHub issue~ N/A

### + If there are functional changes:
